app/apollo/resolvers/subscription.js

```diff
@@ -328,13 +330,41 @@ const subscriptionResolvers = {
         if( clusterId ) validateString( 'clusterId', clusterId );
         tags.forEach( value => { validateString( 'tags', value ); } );
 
+        await validAuth(me, org_id, ACTIONS.CREATE, TYPES.SUBSCRIPTION, queryName, context, [name]);
+        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - subscription authorized`);
+
         const kubeOwnerId = await models.User.getKubeOwnerId(context);
 
         // validate the number of total subscriptions are under the limit
         await validateSubscriptionLimit( org_id, 1, context );
 
         // loads the channel
         const channel = await models.Channel.findOne({ org_id, uuid: channel_uuid });
+        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - found: ${!!channel}`);
+
+        const channelIdentifiers = channel ? [channel_uuid, channel.name] : [channel_uuid];
+        await validAuth(me, org_id, ACTIONS.READ, TYPES.CHANNEL, queryName, context, channelIdentifiers);
+        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - channel authorized`);
+
+        // loads the groups
+        const requestedGroups = await models.Group.find({
+          org_id,
+          $or: [
+            { name: { $in: groups } },
+            { uuid: { $in: groups } }
+          ]
+        }).lean({ virtuals: true });
+        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - found: ${requestedGroups.length}`);
+
+        const allowedGroups = await filterResourcesToAllowed(me, org_id, ACTIONS.READ, TYPES.GROUP, requestedGroups, context);
+        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - allowed: ${allowedGroups.length}`);
+
+        const groupNames = _.map(allowedGroups, group => group.name);
+
+        // Check for errors after validation
+        if( allowedGroups.length < requestedGroups.length ) {
+          throw new NotFoundError(context.req.t('One or more of the passed group uuids were not found'));
+        }
```

Comment:
Previously, calling `getGroupNames` would cause an error to be thrown if the passed groups were not all found.
The new impl only throws the `NotFoundError` if the user doesn't have access to the **_found_** groups.
E.g. if the users passess in the group names `['earth', 'wind', 'fire']` and _NONE_ of those actually exist, the new impl will continue to add the subscription as if **zero** groups were specified.

The same problem occurs with the `editSubscription` impl changes later

Reply:
adding this to accommodate: 
`        if(foundGroups.length < groups.length || allowedGroups.length < groups.length) {
          throw new NotFoundError(context.req.t('One or more of the passed group uuids were not found'));
        }`

Reply:
Marking resolved, but see new comment added in the error checking section